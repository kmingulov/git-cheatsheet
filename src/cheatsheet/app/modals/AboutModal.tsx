import React, { Component, ReactElement } from 'react';
import { Button, Modal } from 'react-bootstrap';

import commands from 'cheatsheet/command-data/commands';

const COMMANDS_COUNT = commands.reduce((counter, group) => counter + group.commands.length, 0);

interface AboutModalProps {
  shown: boolean;
  onClose: () => void;
}

/**
 * Dialog with information about the application.
 * @inheritdoc
 */
export class AboutModal extends Component<AboutModalProps> {
  public shouldComponentUpdate(nextProps: Readonly<AboutModalProps>): boolean {
    return this.props.shown !== nextProps.shown;
  }

  public render(): ReactElement {
    const { shown, onClose } = this.props;

    return (
      <Modal show={ shown } onHide={ onClose }>
        <Modal.Header closeButton>
          <Modal.Title>Git Cheatsheet v{ process.env.REACT_APP_VERSION }</Modal.Title>
        </Modal.Header>

        <Modal.Body>
          <p>
            Git Cheatsheet is an interactive comprehensive cheatsheet which will help you to memorize all these numerous
            git commands!
          </p>

          <p>
            There are currently <b>{ COMMANDS_COUNT }</b> commands in the cheatsheet.
          </p>

          <p>
            Git Cheatsheet is an open-source project, licensed under MIT License and hosted
            on <a href='https://github.com/kmingulov/git-cheatsheet'>GitHub</a>. If you think that the cheatsheet can be
            improved, please fill free to open an issue or submit a pull-request.
          </p>
        </Modal.Body>

        <Modal.Footer>
          <Button variant='primary' onClick={ onClose }>
            Close
          </Button>
        </Modal.Footer>
      </Modal>
    );
  }
}
