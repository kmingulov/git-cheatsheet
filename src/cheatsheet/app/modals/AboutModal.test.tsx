import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import { AboutModal } from './AboutModal';

jest.mock('react-bootstrap', () => {
  const { plainComponent } = require('cheatsheet/util/component-test-utils');

  const Modal = plainComponent('Modal');
  Modal.Header = plainComponent('Modal.Header');
  Modal.Title = plainComponent('Modal.Title');
  Modal.Body = plainComponent('Modal.Body');
  Modal.Footer = plainComponent('Modal.Footer');

  return {
    Button: plainComponent('Button'),
    Modal,
  };
});

jest.mock('cheatsheet/command-data/commands', () => {
  const { Command, CommandGroup } = require('cheatsheet/command');

  return [
    new CommandGroup('basic', 'Basic', [
      new Command('basic-1', [], [], []),
    ]),
    new CommandGroup('advanced', 'Advanced', [
      new Command('advanced-1', [], [], []),
      new Command('advanced-2', [], [], []),
    ]),
  ];
});

Enzyme.configure({adapter: new Adapter()});

describe(AboutModal, () => {
  process.env.REACT_APP_VERSION = '1.0.0';

  it('renders correctly when shown', () => {
    const component = mount(<AboutModal shown={ true } onClose={ () => undefined }/>);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly when hidden', () => {
    const component = mount(<AboutModal shown={ false } onClose={ () => undefined }/>);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly when redrawn', () => {
    const component = mount(<AboutModal shown={ true } onClose={ () => undefined }/>);
    expect(component).toMatchSnapshot();

    component.setProps({ shown: false });
    expect(component).toMatchSnapshot();
  });
});
