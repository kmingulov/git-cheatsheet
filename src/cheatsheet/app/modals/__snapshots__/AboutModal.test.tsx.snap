// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutModal renders correctly when hidden 1`] = `
<AboutModal
  onClose={[Function]}
  shown={false}
>
  <Modal
    onHide={[Function]}
    show={false}
  >
    <Modal.Header
      closeButton={true}
    >
      <Modal.Title>
        Git Cheatsheet v
        1.0.0
      </Modal.Title>
    </Modal.Header>
    <Modal.Body>
      <p>
        Git Cheatsheet is an interactive comprehensive cheatsheet which will help you to memorize all these numerous git commands!
      </p>
      <p>
        There are currently 
        <b>
          3
        </b>
         commands in the cheatsheet.
      </p>
      <p>
        Git Cheatsheet is an open-source project, licensed under MIT License and hosted on 
        <a
          href="https://github.com/kmingulov/git-cheatsheet"
        >
          GitHub
        </a>
        . If you think that the cheatsheet can be improved, please fill free to open an issue or submit a pull-request.
      </p>
    </Modal.Body>
    <Modal.Footer>
      <Button
        onClick={[Function]}
        variant="primary"
      >
        Close
      </Button>
    </Modal.Footer>
  </Modal>
</AboutModal>
`;

exports[`AboutModal renders correctly when redrawn 1`] = `
<AboutModal
  onClose={[Function]}
  shown={true}
>
  <Modal
    onHide={[Function]}
    show={true}
  >
    <Modal.Header
      closeButton={true}
    >
      <Modal.Title>
        Git Cheatsheet v
        1.0.0
      </Modal.Title>
    </Modal.Header>
    <Modal.Body>
      <p>
        Git Cheatsheet is an interactive comprehensive cheatsheet which will help you to memorize all these numerous git commands!
      </p>
      <p>
        There are currently 
        <b>
          3
        </b>
         commands in the cheatsheet.
      </p>
      <p>
        Git Cheatsheet is an open-source project, licensed under MIT License and hosted on 
        <a
          href="https://github.com/kmingulov/git-cheatsheet"
        >
          GitHub
        </a>
        . If you think that the cheatsheet can be improved, please fill free to open an issue or submit a pull-request.
      </p>
    </Modal.Body>
    <Modal.Footer>
      <Button
        onClick={[Function]}
        variant="primary"
      >
        Close
      </Button>
    </Modal.Footer>
  </Modal>
</AboutModal>
`;

exports[`AboutModal renders correctly when redrawn 2`] = `
<AboutModal
  onClose={[Function]}
  shown={false}
>
  <Modal
    onHide={[Function]}
    show={false}
  >
    <Modal.Header
      closeButton={true}
    >
      <Modal.Title>
        Git Cheatsheet v
        1.0.0
      </Modal.Title>
    </Modal.Header>
    <Modal.Body>
      <p>
        Git Cheatsheet is an interactive comprehensive cheatsheet which will help you to memorize all these numerous git commands!
      </p>
      <p>
        There are currently 
        <b>
          3
        </b>
         commands in the cheatsheet.
      </p>
      <p>
        Git Cheatsheet is an open-source project, licensed under MIT License and hosted on 
        <a
          href="https://github.com/kmingulov/git-cheatsheet"
        >
          GitHub
        </a>
        . If you think that the cheatsheet can be improved, please fill free to open an issue or submit a pull-request.
      </p>
    </Modal.Body>
    <Modal.Footer>
      <Button
        onClick={[Function]}
        variant="primary"
      >
        Close
      </Button>
    </Modal.Footer>
  </Modal>
</AboutModal>
`;

exports[`AboutModal renders correctly when shown 1`] = `
<AboutModal
  onClose={[Function]}
  shown={true}
>
  <Modal
    onHide={[Function]}
    show={true}
  >
    <Modal.Header
      closeButton={true}
    >
      <Modal.Title>
        Git Cheatsheet v
        1.0.0
      </Modal.Title>
    </Modal.Header>
    <Modal.Body>
      <p>
        Git Cheatsheet is an interactive comprehensive cheatsheet which will help you to memorize all these numerous git commands!
      </p>
      <p>
        There are currently 
        <b>
          3
        </b>
         commands in the cheatsheet.
      </p>
      <p>
        Git Cheatsheet is an open-source project, licensed under MIT License and hosted on 
        <a
          href="https://github.com/kmingulov/git-cheatsheet"
        >
          GitHub
        </a>
        . If you think that the cheatsheet can be improved, please fill free to open an issue or submit a pull-request.
      </p>
    </Modal.Body>
    <Modal.Footer>
      <Button
        onClick={[Function]}
        variant="primary"
      >
        Close
      </Button>
    </Modal.Footer>
  </Modal>
</AboutModal>
`;
